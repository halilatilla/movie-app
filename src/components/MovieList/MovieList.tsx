import { FC } from 'react'

import { IMovies } from '@src/types'
import MovieCard from '../MovieCard/MovieCard'

interface Props {
  movies: IMovies[]
}

const CharacterList: FC<Props> = ({ movies }) => {
  return (
    <div data-testid="character-list" className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4  gap-8">
      {movies?.map((movie: any) => (
        <MovieCard key={movie.imdbID} movie={movie} />
      ))}
    </div>
  )
}

export default CharacterList
