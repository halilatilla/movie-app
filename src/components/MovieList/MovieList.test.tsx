import { render, screen } from '@testing-library/react'

import MovieList from './MovieList'

describe('Movie List', () => {
  const MOVIES = {
    movies: [],
  }

  it('renders correctly', () => {
    const { container } = render(<MovieList {...MOVIES} />)
    expect(container).toMatchSnapshot()
  })

  it('renders Movie List', () => {
    render(<MovieList {...MOVIES} />)

    const movieList = screen.getByTestId('movie-list')

    expect(movieList).toBeInTheDocument()
  })
})
