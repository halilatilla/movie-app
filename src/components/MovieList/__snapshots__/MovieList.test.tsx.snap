// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List renders correctly 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 xl:grid-cols-4 gap-8"
    data-testid="movie-list"
  />
</div>
`;
